core/mondoo-windows-11-compability.mql.yaml

```diff
@@ -0,0 +1,983 @@
+# Copyright (c) Mondoo, Inc.
+# SPDX-License-Identifier: BUSL-1.1
+policies:
+  - uid: mondoo-windows-11-compatibility
+    name: Windows 11 Compatibility for Workstations
+    version: 1.0.6
+    tags:
+      mondoo.com/platform: windows
+    authors:
+      - name: Martin Souczek
+      - name: Mondoo, Inc
+        email: hello@mondoo.com
+    docs:
+      desc: |
+        The Windows 11 Compatibility Policy checks if a Windows workstation is compatible with Windows 11. The policy checks for the following requirements:
+        - CPU compatibility
+        - RAM requirements
+        - HDD requirements
+        - TPM 2.0 requirements
+    groups:
+      - filters: |
+          asset.platform.contains("windows")
+          windows.computerInfo['OsProductType'] == 1
+        checks:
+          - uid: mondoo-windows-11-compatibility-cpu-list
+          - uid: mondoo-windows-11-compatibility-cpu-cores
+          - uid: mondoo-windows-11-compatibility-cpu-speed
+          - uid: mondoo-windows-11-compatibility-ram
+          - uid: mondoo-windows-11-compatibility-hdd
+          - uid: mondoo-windows-11-compatibility-tpm-present
+    scoring_system: 3
+queries:
+  - uid: mondoo-windows-11-compatibility-cpu-list
+    title: Check Windows 11 CPU compatibility
+    impact: 10
+    props:
+      - uid: ListAllowedCPU
```

Comment:
This is me showing my Windows ignorance, but it this something we can do on capabilities instead of model numbers. The requirements are actually based on CPU capabilities are they not and we if could expose that / check on that we could avoid needing to update this list every few months when a new CPU variant comes out.

Reply:
maybe i am wrong but i found just this list https://learn.microsoft.com/en-gb/windows-hardware/design/minimum/windows-processor-requirements

Reply:
I dug a bit deeper and it looks like this list is not CPU feature based, but instead just totally arbitrary from MS. That means sadly we can't automate this.